src/ConnectedMachine/custom/Connect-AzConnectedMachine.ps1

```diff
@@ -0,0 +1,254 @@
+                $t += "=$($Tag[$key])"
+            }
+            $t
+        }
+
+        $azcmagentArgs.Add([string]::Join(',', $tagStrings))
+    }
+
+    $script = {
+        if ($IsMacOS) {
+            throw "Registration doesn't work on macOS. Only Linux and Windows are supported."
+        }
+
+        if ($IsLinux) {
+            # Download the installation package
+            Invoke-RestMethod -Uri https://aka.ms/azcmagent -OutFile ~/install_linux_azcmagent.sh -UseBasicParsing
+
+            # Install the hybrid agent
+            $installArgs = if ($Proxy) {
+                '--proxy', $P.ToString()
+            } else {
+                @()
+            }
+            bash ~/install_linux_azcmagent.sh @installArgs | ForEach-Object {
+                Write-Host $_
+            }
+
+            # Set executable path
+            $azcmagentPath = "azcmagent"
+        } else {
+            Invoke-RestMethod -Uri https://aka.ms/AzureConnectedMachineAgent -OutFile AzureConnectedMachineAgent.msi -UseBasicParsing
+    
+            # Install the package
+            msiexec /i AzureConnectedMachineAgent.msi /l*v installationlog.txt /qn | ForEach-Object {
+                Write-Host $_
+            }
+
+            if ($Proxy) {
+                [System.Environment]::SetEnvironmentVariable("https_proxy", $Proxy.ToString(), "Machine")
+            }
+
+            # Set executable path
+            $azcmagentPath = "$env:ProgramFiles\AzureConnectedMachineAgent\azcmagent.exe"
+        }
+
+        # Run connect command
+        if ($IsLinux) {
+            sudo $azcmagentPath @args | ForEach-Object {
+                Write-Host $_
+            }
+        } else {
+            & $azcmagentPath @args | ForEach-Object {
+                Write-Host $_
+            }
+        }
+
+        if ($LASTEXITCODE -ne 0) {
+            throw "azcmagent exited with a non-zero exit code: $LASTEXITCODE"
+        }
+
+        $showResult = if ($IsLinux) {
+            sudo $azcmagentPath show
+        } else {
+            # Windows
+            & $azcmagentPath show
+        }
+
+        # Combine the output into one string
+        [string]::Join("`n", $showResult)
+    }
+
+    if ($PSSession) {
+        $scriptWithUsings = {
+            if ($IsLinux) {
+                Write-Error "Linux PSSessions are not supported because PowerShell doesn't support sudo password prompts over PSSessions. (PowerShell issue #1527)"
+                return
+            }
+
+            $Proxy = $using:Proxy
+            $script = $using:script
+
+            Invoke-Command -ScriptBlock ([scriptblock]::Create($script)) -ArgumentList $args
+        }
+        $showResult = Invoke-Command -ScriptBlock $scriptWithUsings -Session $PSSession -ArgumentList $azcmagentArgs
+    } else {
+        $showResult = & $script $azcmagentArgs
+    }
+
+    # Handle show result by getting the name and getting the machine object
+    $showResult | ForEach-Object {
```

Comment:
What is the performance of the cmdlet?

Reply:
`Connect-AzConnectedMachine` takes about 20-30 seconds. The bottle neck is the performance of the actual agent that is downloaded, not the cmdlet.